includes/core-changes/corefx/5.0/obsolete-consoleloggeroptions-properties.md

```diff
@@ -0,0 +1,129 @@
+
+  ```csharp
+  loggingBuilder.AddConsole(options =>
+  {
+    options.FormatterName = ConsoleFormatterNames.Systemd;
+  });
+  ```
+
+  There are several differences between <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.FormatterName> and <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.Format>:
+
+  - <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.Format> has only two possible options: `Default` and `Systemd`.
+  - <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.FormatterName> is case insensitive and can be any string. The reserved, built-in names are `Simple`, `Systemd`, and `Json` (.NET 5.0 and later).
+  - `"Format": "Systemd"` maps to `"FormatterName": "Systemd"`.
+  - `"Format": "Default"` maps to `"FormatterName": "Simple"`.
+
+- For the <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.DisableColors>, <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.IncludeScopes>, <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.TimestampFormat>, and `UseUtcTimestamp` properties, use the corresponding property on the new `Microsoft.Extensions.Logging.Console.JsonConsoleFormatterOptions` or `Microsoft.Extensions.Logging.Console.SimpleConsoleFormatterOptions` types instead. For example:
+
+  ```csharp
+  loggingBuilder.AddSimpleConsole(options =>
+  {
+      options.DisableColors = true;
+  });
+  ```
+
+The following two JSON snippets show how the configuration file changes. Old configuration file:
+
+```json
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "None",
+      "Microsoft": "Warning",
+      "Microsoft.Hosting.Lifetime": "Information"
+    },
+
+    "Console": {
+      "LogLevel": {
+        "Default": "Information"
+      },
+      "Format": "Systemd",
+      "IncludeScopes": true,
+      "TimestampFormat": "HH:mm:ss",
+      "UseUtcTimestamp": true
+    }
+  },
+  "AllowedHosts": "*"
+}
+```
+
+New configuration file:
+
+```json
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "None",
+      "Microsoft": "Warning",
+      "Microsoft.Hosting.Lifetime": "Information"
+    },
+
+    "Console": {
+      "LogLevel": {
+        "Default": "Information"
+      },
+      "FormatterName": "Systemd",
+      "FormatterOptions": {
+        "IncludeScopes": true,
+        "TimestampFormat": "HH:mm:ss",
+        "UseUtcTimestamp": true
+      }
+    }
+  },
+  "AllowedHosts": "*"
+}
+```
+
+#### Category
+
+- Core .NET libraries
+- ASP.NET
+
+#### Affected APIs
+
+- <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.DisableColors?displayProperty=fullName>
+- <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.IncludeScopes?displayProperty=fullName>
+- <xref:Microsoft.Extensions.Logging.Console.ConsoleLoggerOptions.TimestampFormat?displayProperty=fullName>
+- `UseUtcTimestamp`
```

Comment:
What is this type?
Googling it gives no results, and it's the only one without an xref

Reply:
Searching in runtime repo, it's a public property defined [here](https://github.com/dotnet/runtime/blob/345ab7779ddb8ee5a4b0a354d866be27ec7221ac/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerOptions.cs#L65). But why it doesn't exist in API reference?

Reply:
Interesting...thanks for finding that. It looks like the property was [introduced last October](https://github.com/dotnet/runtime/commit/6db6c2783c86ef41a2991fe45bb291647e601a97#diff-efe95832f92d3e580938e385cd1091eb). I'll check if it's in the binary that we created the API docs from.

Reply:
Something's not right here. @safern the Microsoft.Extensions.Logging.Console.dll file is listed under the microsoft.aspnetcore.app folder in the preview 7 assemblies that you sent me. We do not include the assemblies in that folder when generating the API docs. What needs to be changed here? /cc @BillWagner

Reply:
The reason why they're under the `Microsoft.AspNetCore.App` folder is because that ships as part of the ASP.NET shared framework.

cc: @mkArtakMSFT @maryamariyan

Reply:
Okay, so it sounds like we should be copying those assemblies over when we generate the .NET API docs. Will there be any duplication with the ASP.NET API docs if we do that?

Reply:
Maybe we should only copy the Microsoft.Extensions* assemblies from the microsoft.aspnetcore.app folder? /cc @mairaw for any help you can provide too.

Reply:
Part of the extensions was moved into the Platform Extensions but not sure if we have a definitive list (mainly the parts that could be used on other projects not just ASP.NET). I'll forward a thread I have on this.

Reply:
We have a definitive list. Part of extensions ships as both packages and part of Microsfot.AspNetCore.App and the remaining ships as just packages.

Reply:
See https://github.com/dotnet/aspnetcore/issues/19806#issue-580165912

Reply:
This is the list I use... whatever doesn't start with `Microsoft.AspNetCore` is part of Platform Extensions.
https://github.com/dotnet/api-catalog-infra/blob/feb11538b8b19972da8e79d0dbfbee52cb2d34c3/eng/droplayout.props#L63-L104

Reply:
@safern Your list doesn't match up with Sourabh's at all. Also, it doesn't include the Microsoft.Extensions.Logging.Console assembly that's the subject of this missing API. I will plan to import all the APIs from all the assemblies that moved from dotnet/extensions to dotnet/runtime:

- Microsoft.Extensions.Caching.Abstractions
- Microsoft.Extensions.Caching.Memory
- Microsoft.Extensions.Configuration
- Microsoft.Extensions.Configuration.Abstractions
- Microsoft.Extensions.Configuration.Binder
- Microsoft.Extensions.Configuration.CommandLine
- Microsoft.Extensions.Configuration.EnvironmentVariables
- Microsoft.Extensions.Configuration.FileExtensions
- Microsoft.Extensions.Configuration.Ini
- Microsoft.Extensions.Configuration.Json
- Microsoft.Extensions.Configuration.UserSecrets
- Microsoft.Extensions.Configuration.Xml
- Microsoft.Extensions.DependencyInjection
- Microsoft.Extensions.DependencyInjection.Abstractions
- Microsoft.Extensions.FileProviders.Abstractions
- Microsoft.Extensions.FileProviders.Composite
- Microsoft.Extensions.FileProviders.Physical
- Microsoft.Extensions.FileSystemGlobbing
- Microsoft.Extensions.Hosting
- Microsoft.Extensions.Hosting.Abstractions
- Microsoft.Extensions.Http
- Microsoft.Extensions.Logging
- Microsoft.Extensions.Logging.Abstractions
- Microsoft.Extensions.Logging.Configuration
- Microsoft.Extensions.Logging.Console
- Microsoft.Extensions.Logging.Debug
- Microsoft.Extensions.Logging.EventLog
- Microsoft.Extensions.Logging.EventSource
- Microsoft.Extensions.Logging.Testing
- Microsoft.Extensions.Logging.TraceSource
- Microsoft.Extensions.Options
- Microsoft.Extensions.Options.ConfigurationExtensions
- Microsoft.Extensions.Options.DataAnnotations
- Microsoft.Extensions.Primitives

Reply:
My list doesn't match just list because my list doesn't contain the Microsoft.Extensions packages that ship as part of Microsoft.AspNetCore.App package.

It didn't make sense to include those packages as part of both Platform Extensions and aspnetcore.

I can double check though, but I'm pretty sure, cause at the beginning we started including all Microsoft.Extensions.* packages into Platform Extensions and I figured there were duplicates.

Reply:
What we can do so that they appear as part of the docs is to include all the packages that are produced in runtime (Microsoft.Extensions.*) into the platform extensions drop if that's what we want.

Reply:
> What we can do so that they appear as part of the docs is to include all the packages that are produced in runtime (Microsoft.Extensions.*) into the platform extensions drop if that's what we want.

Yes, it sounds to me like that would work going forward. I'm a noob here though.